packages/lib/modules/pool/pool.tokens.display.ts

```diff
@@ -0,0 +1,108 @@
+import { GqlNestedPool, GqlPoolBase } from '@repo/lib/shared/services/api/generated/graphql'
+import { Pool } from './PoolProvider'
+import { ApiToken, ApiTokenWithBalance, PoolCore, PoolListItem } from './pool.types'
+// import { isV3Pool } from './pool.helpers'
+import { PoolToken } from '../tokens/token.helpers'
+import { isSameAddress } from '@repo/lib/shared/utils/addresses'
+import { Address } from 'viem'
+import { sortBy } from 'lodash'
+
+// CRITICAL TODO: duplicated function as importing from pool.helpers creates a circular dependency or something
+// (use the import from pool.helpers to see the error)
+export function isV3Pool(
+  pool: Pick<Pool | PoolListItem | GqlPoolBase, 'protocolVersion'>
+): boolean {
+  return pool.protocolVersion === 3
+}
+
+export function getCompositionDisplayTokens(pool: PoolCore | GqlNestedPool): ApiTokenWithBalance[] {
+  const tokens = getPoolTokens(pool).map(token => {
+    if (token.hasNestedPool && token.nestedPool) {
+      return {
+        ...token,
+        nestedTokens: getCompositionDisplayTokens(token.nestedPool as GqlNestedPool),
+      } as ApiTokenWithBalance
+    }
+    return token as ApiTokenWithBalance
+  })
+
+  return sortApiTokensBySymbol(
+    excludeNestedBptTokens(tokens as ApiToken[], pool.address)
+  ) as ApiTokenWithBalance[]
+}
+
+export function getHeaderDisplayTokens(pool: PoolCore): ApiToken[] {
+  //  excludeNestedBptTokens(pool.poolTokens, pool.address) //TODO: do we need this case?? How is Panthom displayed after API fix?
+  if (isV3Pool(pool) && pool.hasErc4626 && pool.hasAnyAllowedBuffer) {
+    return sortApiTokensBySymbol(
+      pool.poolTokens.map(token =>
+        token.isErc4626 && token.isBufferAllowed
+          ? ({ ...token, ...token.underlyingToken } as ApiToken)
+          : (token as ApiToken)
+      )
+    )
+  }
+
+  return sortApiTokensBySymbol(getCompositionDisplayTokens(pool))
+}
+
+function isPool(pool: any): pool is Pool {
+  return (pool as Pool).poolTokens !== undefined
+}
+
+function isGqlNestedPool(pool: any): pool is GqlNestedPool {
+  return (pool as GqlNestedPool).tokens !== undefined
+}
+
+function getPoolTokens(pool: PoolCore | GqlNestedPool): PoolToken[] {
+  if (isPool(pool)) {
+    return pool.poolTokens
+  }
+  if (isGqlNestedPool(pool)) {
+    return pool.tokens
+  }
+  throw new Error('Invalid pool type: poolTokens or tokens but be defined')
+}
+
+function sortApiTokensBySymbol(tokens: ApiToken[]): ApiToken[] {
+  return sortBy(tokens, 'symbol')
+}
+
+function excludeNestedBptTokens(tokens: PoolToken[] | ApiToken[], poolAddress: string): ApiToken[] {
+  return tokens
+    .filter(token => !isSameAddress(token.address, poolAddress as Address)) // Exclude the BPT pool token itself
+    .filter(token => token !== undefined) as ApiToken[]
+}
+
+export function getHeaderTokensWithPossibleNestedTokensWithBalance(
```

Comment:
I really dislike long function names like this. Should these not be flags to the underlying function?
```
getHeaderTokens(pool, { includeNested: true, includeBalance: true }).
```